integration-test-processor/src/main/java/org/springframework/cloud/stream/module/it/IntegrationTestProcessor.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.it;
+
+import javax.annotation.PostConstruct;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Processor;
+
+/**
+ * A module that can misbehave, useful for integration testing of module deployers.
+ *
+ * @author Eric Bottard
+ */
+@EnableBinding(Processor.class)
+public class IntegrationTestProcessor {
+
+	@Autowired
+	private IntegrationTestProcessorProperties properties;
+
+	@Value("${INSTANCE_INDEX:${CF_INSTANCE_INDEX:0}}")
+	private int instanceIndex;

```

Comment:
Would there be any value in injecting ChannelBindingServiceProperties and getting the instance index from there? Just for the sake of avoiding duplicating the logic there (in case it changes for whatever reason)?